src/map-engine.js

```diff
@@ -0,0 +1,226 @@
+                break;
+            case 'NW' :
+                movementOptionInfos.push(this.obtainMovementOptionInfo(map[player.x-1][player.y-1]));
+                break;
+            case 'SE' :
+                movementOptionInfos.push(this.obtainMovementOptionInfo(map[player.x+1][player.y+1]));
+                break;
+            case 'SW' :
+                movementOptionInfos.push(this.obtainMovementOptionInfo(map[player.x-1][player.y+1]));
+                break;
+        }
+    }
+    for(var i = 0; i < movementOptions.length; i++){
+        console.log(colors.cyan(movementOptions[i]) + colors.dim('('+this.getFullDirection(movementOptions[i])+')')+' : ' + colors.cyan(movementOptionInfos[i]));
+    }
+}
+
+mapEngine.getObjectFromCurrentPosition = function(x,y,direction){
+    switch(direction){
+        case 'S' :
+            return map[x][y+1];
+        case 'N' :
+            return map[x][y-1];
+        case 'E' :
+            return map[x+1][y];
+        case 'W' :
+            return map[x-1][y];
+        case 'NE' :
+            return map[x+1][y-1];
+        case 'NW' :
+            return map[x-1][y-1];
+        case 'SE' : 
+            return map[x+1][y+1];
+        case 'SW' :
+            return map[x-1][y+1];
+        default:
+            return map[x][y];
+    }
+}
+
+mapEngine.movePlayerTowards = function(direction){
+    switch(direction){
+        case 'S' : 
+            player.prev_x = player.x;
+            player.prev_y = player.y++;
+            break;
+        case 'N' : 
+            player.prev_x = player.x;
+            player.prev_y = player.y--;
+            break;
+        case 'E' :
+            player.prev_x = player.x++;
+            player.prev_y = player.y;
+            break;
+        case 'W' :
+            player.prev_x = player.x--;
+            player.prev_y = player.y;
+            break;
+        case 'NE':
+            player.prev_x = player.x++;
+            player.prev_y = player.y--;
+            break;
+        case 'NW':
+            player.prev_x = player.x--;
+            player.prev_y = player.y--;
+            break;
+        case 'SE':
+            player.prev_x = player.x++;
+            player.prev_y = player.y++;
+            break;
+        case 'SW':
+            player.prev_x = player.x--;
+            player.prev_y = player.y++;
+            break;
+    }
+}
+
+mapEngine.updateNextAndPreviousMapTiles = function(x,y,prev_x,prev_y,nextType,nextAquiredBy,prevType,prevAquiredBy){
+    map[x][y].type = nextType;
+    map[x][y].aquiredBy = nextAquiredBy;
+    map[prev_x][prev_y].type = prevType;
+    map[prev_x][prev_y].aquiredBy = prevAquiredBy;
+}
+
+mapEngine.updateNextAndPreviousMapTilesAfterPlayerMovement = function(player,prevType,prevAquiredBy){
+    this.updateNextAndPreviousMapTiles(player.x,player.y,player.prev_x,player.prev_y,'player',player,prevType,prevAquiredBy);
+}
+
```

Comment:
@ChristianRoper There was a problem here because `var map` is not exported. Please add the following before line 226...
```suggestion
mapEngine.map = map;
```
afterwards update `app.js` as mentioned.